### ISSUE TITLE
(integ-runner): (Snapshot serializer to ignore certain fields)

### ISSUE THREAD

**Issue by webratz** [aws/aws-cdk#31299]
### Describe the feature

The snapshots created by the integ-runner contain the full templates. There is certain fields that are changing (eg. tags, stack metadata, asset ids) that should not trigger a new test.
In normal snapshots tests we use for example https://github.com/hupe1980/jest-cdk-snapshot to avoid adding these volatile fields to the snapshot, but I have found no way to do the same for integration tests.

### Use Case

This feature would help us to avoid unneccessary re-runs of integration stacks for unrelated changes.

### Proposed Solution

There already seems to be some logic for the template. A way to hook into that, or extending it, to cover use cases like ignoring metadata, assets or tags.

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.153

### Environment details (OS name and version, etc.)

debian - unrelevant

**Comment by pahud** [comment]
>There is certain fields that are changing (eg. tags, stack metadata, asset ids) that should not trigger a new test.

Yes I basically agree with you on this point. Feel free to submit your PR and the team would take a look when it's ready. Thank you.

**Comment by webratz** [comment]
Do you have any guidance on where would be a good place to implement this? Probably makes sense to briefly align on the basic structure before I have a closer look into this
